notification-infrastructure/src/main/kotlin/io/github/v1servicenotification/global/security/AuthenticationFilter.kt

```diff
@@ -0,0 +1,43 @@
+package io.github.v1servicenotification.global.security
+
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.context.SecurityContextHolder
+import org.springframework.security.core.userdetails.User
+import org.springframework.security.core.userdetails.UserDetails
+import org.springframework.web.filter.OncePerRequestFilter
+import javax.servlet.FilterChain
+import javax.servlet.http.HttpServletRequest
+import javax.servlet.http.HttpServletResponse
+
+class AuthenticationFilter: OncePerRequestFilter() {
+
+    override fun doFilterInternal(
+        request: HttpServletRequest,
+        response: HttpServletResponse,
+        filterChain: FilterChain
+    ) {
+        if (request.getHeader("Request-User-Id") != null && request.getHeader("Request-User-Role") != null && request.getHeader(
+                "Request-User-Authorities"
+            ) != null
+        ) {
+            val userId = request.getHeader("Request-User-Id")
+            val userRole: UserRole = UserRole.valueOf(request.getHeader("Request-User-Role"))
+            val userAuthorities = listOf(request.getHeader("Request-User-Authorities"))
+            val authorities: MutableCollection<SimpleGrantedAuthority> = ArrayList()
+
+            for (userAuthority in userAuthorities) {
+                authorities.add(SimpleGrantedAuthority(userAuthority))
+            }
+
+            authorities.add(SimpleGrantedAuthority(userRole.name))
```

Comment:
```suggestion
            authorities.add(SimpleGrantedAuthority("ROLE_${userRole.name}"))
```
이런식으로 ROLE_붙이는게 어떨까요?